src/mysql/MySQL.chpl

```diff
@@ -0,0 +1,677 @@
+                
+                this._nRows = mysql_num_rows(this._cptr_result);
+            }
+
+            else {
+                writeln("[Error] Error occurred while storing result.");
+                throw new QueryExecutionError();
+                writeln(createStringWithNewBuffer(mysql_error(this._cptr_mysqlconn)));
+            }
+        }
+
+        /*
+        Execute a batch of SQL statements in a single transaction.
+        Commits after their execution, or rolls back if there was an error.
+            :arg statements: an array of :class:`Statement` objects to be executed
+            :type statements: array of :class:`Statement` objects
+        */
+        override proc executeBatch(statements: [] owned Statement) throws {
+            this._connection.beginTransaction();
+            var hasError: bool = false;
+
+            // TODO: memory management
+            for stmt in statements {
+                try {
+                    this.execute(stmt);
+                }
+                catch err: QueryExecutionError {
+                    hasError = true;
+                    this._connection.rollback();
+
+                    // re-throw the query execution error
+                    throw new QueryExecutionError();
+                }
+            }
+
+            if (!hasError) {
+                this._connection.commit();
+            }
+        }
+
+        /*
+        Fetches the next row of the result set.
+        Returns a row whcih gives false on calling isValid() on it.
+        */
+        override proc fetchone(): owned MySQLRow? {
+            if (this._curRow >= this._nRows) {
+                return nil;
+            }
+
+            var nextRow: MYSQL_ROW = mysql_fetch_row(this._cptr_result);
+            // TODO: is there any way to check if nextRow is NULL here?
+            // using nextRow == nil or nextRow == c_nil don't work
+
+            this._curRow += 1;
+
+            // init the row:
+            var _row = new MySQLRow(nextRow, this._cptr_fields, true);
+            
+            return _row;
+        }
+
+        /*
+        Fetches the next given number of rows in the result one by one,
+        returning an iterator.
+        Throws Error if howManyRows is negative.
+
+            :args howManyRows: how many rows to return
+            :type howManyRows: int(32)
+        */
+        override iter fetchsome(howManyRows: int(32)): owned MySQLRow? {
+            if (howManyRows < 0) {
+                writeln("[Error] MySQLCursor.fetchsome(howManyRows) called with howManyRows < 0");
+                throw new Error();
+            }
+
+            else {
+                var counter = 0;
+                var _row = this.fetchone();
+
+                while (_row != nil && counter < this._nRows && counter < howManyRows) {
+                    yield _row;
+                    _row = this.fetchone();
+                    counter += 1;
+                }
+            }
+        }
```

Comment:
Don't you think there should be an error if `howManyRows` exceeds `nRows`

Reply:
`howManyRows <= _nRows` is a precondition for the function (should include it in the docs though, thanks for noticing it), and it will return rows only up to the actual number of rows (`_nRows`) in the retrieved dataset even if `howManyRows` exceeds `_nRows`; so maybe an error itself is not necessary (seems too severe in my opinion). Rather, I think it would be better to mention about this in the docstring of the function (saying that if `hawManyRows > actual no. of rows`, the function returns all the rows), and possibly, display a warning if this happens (to aid debugging). 

It should also be kept in mind that the user does not directly have the capability to find the number of rows in the fetched set (unless they use `_nRows` itself). 
The reason I left a check for `howManyRows < 0` though is because this would cause the loop itself to malfunction (it won't execute even once).
How does that sound?